src/components/userFeedback/UserFeedback.spec.tsx

```diff
@@ -0,0 +1,134 @@
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.VALID, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.WARNING, 'm1', 'block'))).not.toThrow();
+        expect(() => getShallowOutput(generateProps('hello', UserFeedbackState.ERROR, 'm1', 'block'))).not.toThrow();
+      });
+    });
+
+    describe('text content', () => {
+      it('should contain the exact text passed to prop feedbackText', () => {
+        let testText = 'testing';
+        let wrongText: string;
+
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).toBe(testText);
+
+        testText = 'hello this is a test over here';
+        wrongText = 'hello this is the wrong text over here';
+        expect(getShallowOutput(generateProps(testText, '', '')).text()).not.toBe(wrongText);
+      });
+    });
+
+    describe('style for each state', () => {
+      describe('non existent state', () => {
+        it('should be invisible if state provided does not exist', () => {
+          let nonExistentState = 'NON_EXISTENT_STATE';
+          let emptyState = '';
+
+          expect(getShallowOutput(generateProps('', nonExistentState, '')).hasClass('hidden')).toBe(true);
+          expect(getShallowOutput(generateProps('', emptyState, '')).hasClass('hidden')).toBe(true);
+        });
+      });
+
+      describe('VALID state', () => {
+        it('should be invisible on state VALID without displayOnShow prop', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '')).hasClass('hidden')).toBe(true);
+        });
+
+        it('should be invisible on state VALID, even with prop displayOnShow provided', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '', 'block')).hasClass('hidden')).toBe(true);
+          expect(getShallowOutput(generateProps('', UserFeedbackState.VALID, '', 'block')).hasClass('block')).toBe(false);
+        });
+      });
+
+      describe('WARNING state', () => {
+        it('should be visible (without prop displayOnShow)', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, '')).hasClass('block')).toBe(true);
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, '')).hasClass('hidden')).toBe(false);
+        });
+
+        it('should have a display class equal to the value of the displayOnShow prop', () => {
+          expect(getShallowOutput(generateProps('', UserFeedbackState.WARNING, 'inline-block')).hasClass('inline-block')).toBe(true);
```

Comment:
Hey Germain, I don't know if you meant like real enums like specified in the TS docs, but I could not comprehend how it would apply to our context. I think string literal types https://www.typescriptlang.org/docs/handbook/advanced-types.html would have been great but we can not use them in TS 1.7 (as JP pointed out to me). That being said I know Steph refered to enum as something like 

```
export const DisplayClass = {
  BLOCK: 'block',
  INLINE: 'inline',
  INLINE_BLOCK: 'inline-block',
  HIDDEN: 'hidden',
  TABLE: 'table',
  TABLE_CELL: 'table-cell'
};
``` 

which I implemented. Does that represent added value to you? I've put it in a new file called ComponentUtils.ts under the utils folder, all components can use this enum now. In jsadmin-v2 you would then do ReactVapor.DisplayClass.YOURCHOSENDISPLAYCLASS